nameresolution/kubernetes/kubernetes.go

```diff
@@ -45,10 +45,13 @@ func (k *resolver) Init(metadata nameresolution.Metadata) error {
 	if err != nil {
 		return err
 	}
-	if config, ok := configInterface.(map[string]string); ok {
-		clusterDomain := config[ClusterDomainKey]
-		if clusterDomain != "" {
-			k.clusterDomain = clusterDomain
+	if config, ok := configInterface.(map[string]interface{}); ok {
+		clusterDomainPtr := config[ClusterDomainKey]
+		if clusterDomainPtr != nil {
+			clusterDomain, _ := clusterDomainPtr.(string)
+			if clusterDomain != "" {
```

Comment:
Could we add this case to test?

Reply:
I change the test case, the bug is because when the dapr initialize the nameResolution resolver it pass the metadata.Configuration as map[string]interface{} instead of map[string]string